### PR TITLE
Allow configuring a custom marshaller

### LINKED ISSUE #9: Only JSON messages supported
Currently marshalling to JSON is baked into the library.

Would you accept a PR where we allow overriding this so that other content types can be used?

### PR DESCRIPTION
**Pull request by rubenv**
Fixes #9

Something like this should do the trick without breaking any existing code/behavior. Open to any feedback.

### CODE REVIEW

**Comment by rubenv** [comment]
Rebased on top of current `v1` branch.

File: marshalling.go [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+			case string:
+				return []byte(s), nil
+			case []byte:
+				return s, nil
+			default:
+				return nil, fmt.Errorf("cannot marshal %T as text", data)
+			}
+		},
+	}
+}
```
Comment (m3talux) [inline marshalling.go:47]: Could we add unit tests for the following file? Just to make sure that the marshal method works as intended.

File: client_options.go [hunk 2]
```diff
@@ -195,3 +199,10 @@ func (c *ClientOptions) SetMode(mode string) *ClientOptions {
 
 	return c
 }
+
+// SetMarshaller will assign the Marshaller.
+func (c *ClientOptions) SetMarshaller(marshaller Marshaller) *ClientOptions {
+	c.Marshaller = marshaller
+
+	return c
+}
```
Comment (m3talux) [inline client_options.go:208]: We should probably document this change somewhere in the `README.md` file.

File: manager_options.go [hunk 3]
```diff
@@ -126,3 +130,10 @@ func (m *ManagerOptions) SetMode(mode string) *ManagerOptions {
 
 	return m
 }
+
+// SetMarshaller will assign the Marshaller.
+func (m *ManagerOptions) SetMarshaller(marshaller Marshaller) *ManagerOptions {
+	m.Marshaller = marshaller
+
+	return m
+}
```
Comment (m3talux) [inline manager_options.go:139]: This too, should be documented

File: marshalling.go (hunk 1 above)
Comment (rubenv) [reply marshalling.go:47]: Will do.

File: client_options.go (hunk 2 above)
Comment (rubenv) [reply client_options.go:208]: Not sure what you want me to document here? I've made the change such that even with no marshaller set, the behavior will stay unchanged compared to current behavior.

As such this is entirely optional and won't break any existing code.

File: marshalling.go (hunk 1 above)
Comment (rubenv) [reply marshalling.go:47]: Test added!

File: client_options.go (hunk 2 above)
Comment (m3talux) [reply client_options.go:208]: True, I'll just add a small section for end-users to know that such feature exists before tagging a new version.

**Review by m3talux — APPROVED** [review]

**Pull request merged**